lib/attrsets.nix

```
@@ -523,6 +523,36 @@ rec {
     listToAttrs (map (attr: f attr set.${attr}) (attrNames set));
 
 
+  /* Like mapAttrs, but additionally taking a variable that
+     will be automatically incremented, starting from 0.
```

I think each function should have its own description rather than relying on others. In this case it would be particularly annoying because `mapAttrs` isn't even documented in Nixpkgs (but rather just Nix). This should also be fixed at some point.

```suggestion
  /*
    Map the values of an attribute set to result of calling a function on each.
    The function gets three positional arguments:
    1. (Int) The attribute's alphabetical index, starting from `0`
    2. (String) The attribute's name
    3. (Any) The attribute's value
    
    This function is similar to [`builtins.mapAttrs`](https://nixos.org/manual/nix/stable/language/builtins.html#builtins-mapAttrs),
    but additionally passes the index to the function.
```

(also by formatting it this way, you can have even indentation in the entire doc comment!)

Same for `imapAttrs'`